File: src/test/java/it/unibo/crabinv/Model/i18n/TestLocalization.java
package it.unibo.crabinv.Model.i18n;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class TestLocalization {
    private Localization loc;

    @BeforeEach
    public void setup() {
        loc = new Localization(SUPPORTED_LOCALES.ENGLISH);
    }

    @Test
    void testCorrectKeyFetch() {
        String expectedResult = "PLAY";
        Assertions.assertEquals(expectedResult, loc.getString(TextKeys.PLAY));
    }

    @Test
    void testCorrectLocaleSwap() {
        String expectedResult = "GIOCA";
        loc.setLocale(SUPPORTED_LOCALES.ITALIAN);
        Assertions.assertEquals(expectedResult, loc.getString(TextKeys.PLAY));
    }

    @Test
    void testCorrectLocaleSwapped() {
        var expectedResult = SUPPORTED_LOCALES.ITALIAN;
        loc.setLocale(SUPPORTED_LOCALES.ITALIAN);
        Assertions.assertEquals(expectedResult, loc.getCurrentLocale());
        expectedResult = SUPPORTED_LOCALES.ENGLISH;
        loc.setLocale(SUPPORTED_LOCALES.ENGLISH);
        Assertions.assertEquals(expectedResult, loc.getCurrentLocale());
    }
}

File: src/test/java/it/unibo/crabinv/model/core/i18n/TestLocalization.kt
package it.unibo.crabinv.model.core.i18n

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class TestLocalization {
    private var loc: Localization? = null

    @BeforeEach
    fun setup() {
        loc = Localization(SupportedLocales.ENGLISH)
    }

    @Test
    fun testCorrectKeyFetch() {
        val expectedResult = "PLAY"
        Assertions.assertEquals(expectedResult, loc!!.getString(TextKeys.PLAY))
    }

    @Test
    fun testCorrectLocaleSwap() {
        val expectedResult = "GIOCA"
        loc!!.setLocale(SupportedLocales.ITALIAN)
        Assertions.assertEquals(expectedResult, loc!!.getString(TextKeys.PLAY))
    }

    @Test
    fun testCorrectLocaleSwapped() {
        var expectedResult = SupportedLocales.ITALIAN
        loc!!.setLocale(SupportedLocales.ITALIAN)
        Assertions.assertEquals(expectedResult, loc!!.getCurrentLocale())
        expectedResult = SupportedLocales.ENGLISH
        loc!!.setLocale(SupportedLocales.ENGLISH)
        Assertions.assertEquals(expectedResult, loc!!.getCurrentLocale())
    }
}
